daterangepicker.js

```diff
@@ -1260,10 +1261,16 @@
             var cal = $(e.target).parents('.calendar');
             var date = cal.hasClass('left') ? this.leftCalendar.calendar[row][col] : this.rightCalendar.calendar[row][col];
 
-            if (this.endDate && !this.container.find('input[name=daterangepicker_start]').is(":focus")) {
-                this.container.find('input[name=daterangepicker_start]').val(date.format(this.locale.format));
-            } else if (!this.endDate && !this.container.find('input[name=daterangepicker_end]').is(":focus")) {
-                this.container.find('input[name=daterangepicker_end]').val(date.format(this.locale.format));
+            if(this.firstDateNoChangeMode){
+                if(!this.endDate){
+                    this.container.find('input[name=daterangepicker_end]').val(date.format(this.locale.format));
+                }
+            }else{
+                if (this.endDate && !this.container.find('input[name=daterangepicker_start]').is(":focus")) {
```

Comment:
Although I see your point, but it doesn't make sense to have Frankenstein code. Let the author figure out that we rewrote his code to better.